view/frontend/templates/button_product_page.phtml

```diff
@@ -154,12 +154,16 @@ if (!$block->isSupportableType()) return;
                         return false;
                     }
 
-                    var validator = $( '#product_addtocart_form' ).validate();
-                    if ( !validator.checkForm() ) {
-                        validator.showErrors();
+                    // wait for validation module init
+                    try {
+                        $('#product_addtocart_form').validation('isValid');
```

Comment:
Would be better to use it both check: `validation()` and `validation('isValid')`.
```
var dataForm = $('#product_addtocart_form');
if(dataForm.validation() && dataForm.validation('isValid')) {
...
}
```

Reply:
Also i would be added a disable button functionality like
`$('#product-page-checkout').attr("disabled", true);`
and 
`$('#product-page-checkout').attr("disabled", false);`

In the example the `$('#product-page-checkout')` is Bolt PPC button

Reply:
> Would be better to use both checks: `validation()` and `validation('isValid')`.
> 
> ```
> var dataForm = $('#product_addtocart_form');
> if(dataForm.validation() && dataForm.validation('isValid')) {
> ...
> }
> ```

@YevhenShBolt I used something like that at first and then realized that .validation() is inherent in .validation('isValid') so it is redundant

Reply:
> Also i would be added a disable button functionality like
> `$('#product-page-checkout').attr("disabled", true);`
> and
> `$('#product-page-checkout').attr("disabled", false);`
> 
> In the example the `$('#product-page-checkout')` is Bolt PPC button

@YevhenShBolt there was a long discussion in Github and Asana with @daisy1754 regarding hard-coding CSS in the plugin core files, especially the "disabled". The conclusion is not to use it and remove it from our CSS eventually. We managed to do it strictly via configuration for the regular checkout, but to do here we would need to add more config optioins (eg. additional javascript for ppc) and include them in the template. It can probably wait for the next release. I agree it would be a better user experience. But let's just fix the bug for now. Btw, to run into the issue, you must be really quick in clicking the button on page load.

Reply:
> Btw, to run into the issue, you must be really quick in clicking the button on page load.
often on slow internet connection or through mobile i can easily click on button, except Magento's button add to cart - it is disabled .

Reply:
CC: @danac-gs

Reply:
@YevhenShBolt I agree 100%. Created a task for myself to have this done for the next release.